### ISSUE TITLE
docker differences as of November 2022

### ISSUE THREAD

**Issue by edsantiago** [containers/podman#16543]
Gotta get my issues filed before we start enforcing the "podman info output" requirement!

This is a followup to #14917 and #16430, the reason being that it's just too hard to track a page with lots of edits: after a certain point the blue checkmarks visually overwhelm the non-checked boxes. Let's just start clean every once in a while. This is the list from `hack/xref-docker-options` as of 2022-11-07, incorporating the edits made in #16430 up through 2022-11-17.

- [ ] docker --tls
- [ ] docker --tlscacert
- [ ] docker --tlscert
- [ ] docker --tlskey
- [ ] docker --tlsverify
- [ ] docker builder
- [ ] docker config
- [ ] docker container cp --follow-link
- [ ] docker container cp -L
- [ ] docker container create --domainname  #15200
- [ ] docker container create --link-local-ip
- [ ] docker container create --volume-driver
- [ ] docker container run --domainname  #15200
- [ ] docker container run --link-local-ip
- [ ] docker container run --volume-driver
- [ ] docker container update --pids-limit
- [ ] docker container update --restart
- [ ] docker cp --follow-link
- [ ] docker cp -L
- [ ] docker create --domainname  #15200
- [ ] docker create --link-local-ip
- [ ] docker create --volume-driver
- [ ] docker image import --platform
- [ ] docker image push --all-tags
- [ ] docker image push -a
- [ ] docker import --platform
- [ ] docker network connect --driver-opt
- [ ] docker network connect --link-local-ip
- [ ] docker network create --attachable
- [ ] docker network create --aux-address
- [ ] docker network create --config-from
- [ ] docker network create --config-only
- [ ] docker network create --ingress
- [ ] docker network create --ipam-opt
- [ ] docker network create --scope
- [ ] docker network inspect --verbose
- [ ] docker network inspect -v
- [ ] docker push --all-tags
- [ ] docker push -a
- [ ] docker run --domainname  #15200
- [ ] docker run --link-local-ip
- [ ] docker run --volume-driver
- [ ] docker secret create --template-driver
- [ ] docker trust  #16463
- [ ] docker update --pids-limit
- [ ] docker update --restart
- [ ] docker version --kubeconfig

**Comment by jakecorrenti** [comment]
I'll take on `docker container update --pids-limit`

**Comment by rhatdan** [comment]
You got it.

**Comment by jakecorrenti** [comment]
@edsantiago I forgot to update a commit message and accidentally closed this. I'm sorry

**Comment by jakecorrenti** [comment]
I'll take `docker push -a`

**Comment by csschwe** [comment]
the env option is not currently supported for `podman build --secret`

`docker build --secret id=mysecret,env=MYSECRET`

**Comment by rhatdan** [comment]
@csschwe Please open a new issue on that.  This is just a collection of missing options, if an option exists but does not work correctly then open a new issue.

**Comment by kwilczynski** [comment]
I am unsure if this is the right place or whether I should open a dedicated issue. However, could you also consider adding limits (especially file descriptor limits/fileno) to get the parity between Docker and Podman?

Why? There is a difference between hard/soft fileno for **RUN** vs **CMD** (and **ENTRYPOINT**) in Podman, which is not the case for Docker that seems to be setting **1048576** across the board nowadays, for better or worse, and this seems to trip a lot of users, especially whom are heavily invested in the Red Hat ecosystem (e..g, Podman, Buildah, OpenShift, etc.).

@rhatdan, let me know if this is something you would even consider, and if so, then if you will have some time to work on this or if you would be fine with me sending a Pull Request as a start.

Thank you!

**Comment by rhatdan** [comment]
Please open a separate issue on this.

**Comment by jakecorrenti** [comment]
I would like to give `docker cp --follow-link` a shot

**Comment by rhatdan** [comment]
You got it.

**Comment by danishprakash** [comment]
@jakecorrenti jfyi, that's already being worked upon and discussed [here](https://github.com/containers/podman/issues/16585).

**Comment by jakecorrenti** [comment]
@danishprakash thanks for pointing that out. Since you're working on it already I'll choose something else to work on

**Comment by jakecorrenti** [comment]
I will give `docker update --restart` a shot instead. Doesn't seem like anyone is working on that currently.

**Comment by kwilczynski** [comment]
@rhatdan, I have another potential candidate on the above list.

Would you consider adding an item to ensure that when `podman build --label` or `LABEL` (in a Dockerfile) is used, it has the same behaviour as Docker, which is to add an empty layer to indicate a change?

We recently had a problem with labels and Quay, and it took us a bit to realise that Buildah has different behaviour. That said, I am unsure if this (as in the compatibility issue) belongs here or to Buildah?

**Comment by rhatdan** [comment]
Please open an issue on this on Buildah.

**Comment by edsantiago** [comment]
Closing.  Should there be any docker differences worth noting, we will rely on users to file issues.
